Quick links in plots tab

Luke-Sikina opened the pull request:
Fix cBioPortal/cbioportal#6651

Describe changes proposed in this pull request:
- Created quick plot pills in UI
- Created logic for all pills to determine when they should show up
- Tested all logic for determining which pills are displayed
- Updated some types to make my life easier

# Checks
- [x] Has tests
- [x] The commit log is comprehensible. 

# Any screenshots or GIFs?
![example-1](https://user-images.githubusercontent.com/20069833/66954520-8f02d480-f02e-11e9-87ae-0bf1070e3a32.gif)
![example-2](https://user-images.githubusercontent.com/20069833/66954522-91652e80-f02e-11e9-8d8b-23cb4097eae0.gif)
![example-3](https://user-images.githubusercontent.com/20069833/66954525-93c78880-f02e-11e9-820c-b5fcd22e28fd.gif)
![example-4](https://user-images.githubusercontent.com/20069833/66954528-9629e280-f02e-11e9-88c8-36863ba53ef2.gif)

alisman commented on src/pages/resultsView/plots/PlotsTab.tsx:
```diff
@@ -160,6 +161,9 @@ export interface IPlotsTabProps {
     store:ResultsViewPageStore;
 };
 
+export type Option = {value:string, label:string};
```
Could conflict pretty quickly with other Option types. Maybe more specific name?

Luke-Sikina replied:
Renamed to `PlotsTabOption`

alisman commented on src/pages/resultsView/plots/PlotsTab.tsx:
```diff
@@ -160,6 +161,9 @@ export interface IPlotsTabProps {
 };
 
+export type Option = {value:string, label:string};
+export type DataSource = {[dataType:string]:{value:string, label:string}[]};
```
ditto here.  if it's exported and not meant for general use, maybe give it a more specific name

Luke-Sikina replied:
Renamed to `PlotsTabDataSource`

alisman commented on src/pages/resultsView/plots/QuickPlots.ts:
```diff
@@ -0,0 +1,344 @@
+        dataTypes: Option[],
+        dataSources?: DataSource,
+    ) => ButtonInfo,
+    validate: (
```
i'm not sure validate is the right name.  isApplicableToQuery?

alisman commented on src/pages/resultsView/plots/QuickPlots.ts:
```diff
@@ -0,0 +1,344 @@
+ * the quickPlots constant.
+ */
+type QuickPlot = {
+    process: (
```
more specific name for process

Luke-Sikina replied:
`toButtonInfo`

alisman commented on src/pages/resultsView/plots/PlotsTab.tsx:
```diff
@@ -281,6 +285,69 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
+            !this.props.store.mutations.isComplete
+        ) {
+            return (
+                <div className="quickPlotsContainer">
```
would like to see the wrapper container div in only once place and contents (loader or UI) renderend conditionally inside it.

alisman commented on src/pages/resultsView/plots/PlotsTab.tsx:
```diff
@@ -2371,6 +2423,7 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
                 <div className={'tabMessageContainer'}>
                     <OqlStatusBanner className="plots-oql-status-banner" store={this.props.store} tabReflectsOql={false} />
                     <AlterationFilterWarning store={this.props.store} isUnaffected={true}/>
+                    {this.quickConfigurations}
```
should be inside the plotsTab div, not up here in message container.

alisman replied:
this will also allow us to tighten up space under the buttons.  should the same as other places we use the pills-nav 
![image](https://user-images.githubusercontent.com/186521/67040725-0f3e3e00-f0f2-11e9-8b0e-08cb312d43ce.png)

Luke-Sikina replied:
![Screenshot from 2019-10-17 16-24-20](https://user-images.githubusercontent.com/20069833/67044612-a14a4480-f0fa-11e9-9c77-904a98b79491.png)

alisman commented:
One more aesthetic tweak before we share this: I think the nav-pills should have 8px padding between them.  Like so:

![image](https://user-images.githubusercontent.com/186521/67045888-26365d80-f0fd-11e9-83a4-a9985222f36d.png)

alisman commented:
And one more thing!  When you select a quick link, the quick links briefly go back into loading state.  This shouldn't be necessary.  Lets try to fix it.

jjgao commented:
@Luke-Sikina is it ready for product review? We are all excited about the feature.

Luke-Sikina commented:
> And one more thing! When you select a quick link, the quick links briefly go back into loading state. This shouldn't be necessary. Lets try to fix it.

Fixed


> One more aesthetic tweak before we share this: I think the nav-pills should have 8px padding between them. Like so:
> 
> ![image](https://user-images.githubusercontent.com/186521/67045888-26365d80-f0fd-11e9-83a4-a9985222f36d.png)

Fixed:
![Screenshot from 2019-10-18 10-09-05](https://user-images.githubusercontent.com/20069833/67101366-807ffe80-f18f-11e9-8e04-ce5f8f8ce504.png)

alisman commented on src/pages/resultsView/plots/styles.scss:
```diff
@@ -166,13 +166,23 @@
+      display: flex;
+      margin-right: 5px;
+
+      .plots-tab-pills {
```
i actually think we should apply this  padding to all the nav pills (used a fair amount in app)
